grpc_set_ssl_roots_override_callback for Python

nathanielmanistaatgoogle opened the pull request:
Not yet passing interop tests but ready to at least begin review.

Since this is my first Cython change @robertwb has graciously agreed to also review.

grpc-jenkins commented:
Can one of the admins verify this patch?

grpc-jenkins commented:
Can one of the admins verify this patch?

nathanielmanistaatgoogle commented:
@soltanmm and @robertwb: after changing `cimport libc` to `from libc.string cimport memcpy` and `libc.memcpy` to `memcpy` this now passes `run_interop_tests.py --use_docker -lpython --cloud_to_prod` and I believe is ready for review.

soltanmm commented:
Does the 'now' in

> this now passes

imply that it wasn't passing before?

If so, that's a little baffling to me (but lgtm in any case).

nathanielmanistaatgoogle commented:
@soltanmm: yes. Note the `--cloud_to_prod` flag, which wasn't included in what we were running Friday morning.

robertwb commented on src/python/grpcio/grpc/_cython/_cygrpc/security.pxd.pxi:
```diff
@@ -0,0 +1,34 @@
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cimport cpython
```
Why are you cimporting cpython here?

nathanielmanistaatgoogle replied:
This file started life as a fork of credentials.pxd.pxi, so... if this cimport doesn't need to be here, I'm happy to take it out.

robertwb commented on src/python/grpcio/grpc/_cython/_cygrpc/security.pyx.pxi:
```diff
@@ -0,0 +1,45 @@
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cimport cpython
```
same

nathanielmanistaatgoogle replied:
Analogous reason here - this file started life as a fork of credentials.pyx.pxi. But if there's no reason for the cimport I'm happy to remove it.

nathanielmanistaatgoogle commented:
I've now dropped the `cimport cpython` from my two new files and changed the `.gitignore` file to match the moved credentials location; please quickly rereview.

soltanmm commented:
You found a `Cloak of LGTM (+1)`!

Would you like to equip it now? *(yes/no)*   ![blinking-cursor](http://vignette2.wikia.nocookie.net/uncyclopedia/images/b/b8/Thin_cursor.gif/revision/latest?cb=20091129041432)

jtattermusch commented:
Does this work when you remove make install-certs  from the python interop docker image?

nathanielmanistaatgoogle commented:
@jtattermusch: Yes! Would you like a follow-on pull request removing that line? I apologize that I forgot to include it in this change.

jtattermusch commented:
Yes, pls send a PR. Thanks!

nathanielmanistaatgoogle commented:
Sent #5168.
